saas/sdk/app/auth.py

```diff
@@ -245,19 +254,49 @@ def update_user(cls, login: str, is_admin: bool, password: Tuple[str, str] = Non
                     name=record.name,
                     disabled=record.disabled,
                     hashed_password=record.hashed_password,
-                    keystore=cls._resolve_keystore(record.keystore_id, record.keystore_password)
+                    keystore=cls._resolve_keystore(record.keystore_id, record.keystore_password),
+                    login_attempts=record.login_attempts
                 )
 
             else:
                 raise AppRuntimeError("Username does not exist", details={
                     'login': login
                 })
 
+    @classmethod
+    def update_login_attempts(cls, login: str, successful: bool) -> User:
+        with cls._Session() as session:
+            # check if this username already exists
+            user = session.query(UserRecord).get(login)
+
+            if user:
+                # reset the login attempts count, if it's a successful login
+                if successful:
+                    user.login_attempts = 0
+                # increase the login attempts count, if it's a failed login attempt
+                else:
+                    user.login_attempts += 1
+                session.commit()
+
+                return User(
```

Comment:
Why return the user object even if the login attempt was unsuccessful?

Reply:
Maybe it requires some refactoring whereby one can get a `User` object and then modify its properties which are then persisted to the db. At the moment we have a few class methods that make changes and then return a `User` object.

Reply:
Sorry, I just realised that the comments here are actually referring to the login attempts PR, not the removal of unnecessary secret key use. Please make changes in the other branch accordingly...